File: WCFPRIMAACSSYS/WCFPRIMAACSSYS/IPrimavera_Accsys.vb
Imports System.ServiceModel

' NOTE: You can use the "Rename" command on the context menu to change the interface name "IPrimavera_Accsys" in both code and config file together.
<ServiceContract()>
Public Interface IPrimavera_Accsys

    ''' <summary>
    ''' Esta operação será responsável por inscrever um aluno através do método inscreverAluno(inputs listados); 
    '''  Será necessário o preenchimento dos campos obrigatórios;
    '''  Será necessário a validação dos campos, de forma a evitar que o utilizador insira dados errados;
    '''  Depois do preenchimento dos dados, deverá existir um botão Gravar;
    '''  A informação deverá se encontrar na base de dados do Fenix, a espera de serem válidados;
    '''  Só depois de serem válidados pela SA, esta informação será carregada no Primavera através de um botão Validar.
    ''' NOTA1: O campo “bolsa”, será do tipo boolean, essa informação depois será manipulada manualmente no Primavera;
    ''' NOTA2: O campo “status”, é que vai definir se ocorreu um erro ou não em relação a comunicação Fenix-Primavera. Onde só depois do Primavera confirmar que gravou o estudante, este campo irá retornar um “ok”, caso não permanecerá com “erro”.
    ''' </summary>
    ''' <param name="nrEstudante"></param>
    ''' <param name="nomeCompleto"></param>
    ''' <param name="nrTalao"></param>
    ''' <param name="morada"></param>
    ''' <param name="bairro"></param>
    ''' <param name="nrTelefone"></param>
    ''' <param name="nrTelefone2"></param>
    ''' <param name="bosla"></param>
    ''' <param name="curso"></param>
    ''' <param name="emailIsutc"></param>
    ''' <param name="emailAlternativo"></param>
    ''' <param name="turma"></param>
    ''' <param name="tipoIngresso"></param>
    ''' <param name="status"></param>
    ''' <param name="observacao"></param>
    ''' <param name="dataRegistro"></param>
    ''' <returns>PrimaveraResultStructure.statusCode</returns>
    ''' <remarks> 
    ''' </remarks>
    <OperationContract(Name = "Inscricao")>
        <WebInvoke(Method = "GET", )>
    Function Inscricao(nrEstudante As String, nomeCompleto As String, ByRef nrTalao As Long, morada As String, bairro As String, _
                       nrTelefone As String, ByRef nrTelefone2 As String, bosla As Boolean, curso As String, emailIsutc As String, _
                       ByRef emailAlternativo As String, turma As String, tipoIngresso As String, ByRef status As String, _
                       ByRef observacao As String, dataRegistro As Date) As String

End Interface

File: WCFPRIMAACSSYS/PRIMAVERA_ACCSYS_WCF/IPrimavera_Accsys.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Web;
using System.Text;

namespace PRIMAVERA_ACCSYS_WCF
{
    [ServiceContract]
    [ServiceKnownType(typeof(PrimaveraResultStructure))]
    //[ServiceContract (Name= "Fenix-Primavera")]
    public interface IPrimavera_Accsys
    {
        [OperationContract(Name = "Operacao_1_Inscricao")]
        [WebInvoke(Method = "GET", ResponseFormat = WebMessageFormat.Xml,
            BodyStyle = WebMessageBodyStyle.Bare,
                UriTemplate = "Inscricao/{nrEstudante},{nomeCompleto}, {morada},{bairro},{nrTelefone},{nrTelefone2},{bolsa}"
                + ",{curso},{emailPessoal},{emailAlternativo},{turma},{tipoIngresso},{status},{observacao}")]
        PrimaveraResultStructure Inscricao(string nrEstudante, string nomeCompleto, string morada, string bairro,
            string nrTelefone, string nrTelefone2, string bolsa, string emailPessoal, string emailAlternativo,
            string tipoIngresso, string status, string observacao);
               
    }

    [DataContract]
    public class PrimaveraResultStructure
    {
        [DataMember(Name = "tipoProblema")]
        public string tipoProblema { get; set; }

        [DataMember(Name = "codigo")]
        public int codigo { get; set; }

        [DataMember(Name = "codeLevel")]
        public string codeLevel { get; set; }

        [DataMember(Name = "subNivel")]
        public string subNivel { get; set; }

        [DataMember(Name = "descricao")]
        public string descricao { get; set; }

        [DataMember(Name = "procedimento")]
        public string procedimento { get; set; }

        [DataMember(Name = "ficheiro")]
        public Stream ficheiro { get; set; }

    }
}
